pkg/operator/controller/ingress/controller_test.go

```diff
@@ -458,6 +567,30 @@ func TestSetDefaultPublishingStrategyHandlesUpdates(t *testing.T) {
 			expectedIC:              makeIC(spec(elb()), status(elbWithNullParameters())),
 			domainMatchesBaseDomain: true,
 		},
+		{
+			name:                    "loadbalancer type changed from NLB to unset, with Classic LB as default",
+			ic:                      makeIC(spec(eps(lbs(operatorv1.ExternalLoadBalancer, &managedDNS))), status(nlb())),
+			ingressConfig:           ingressConfigWithDefaultClassicLB,
+			expectedResult:          true,
+			expectedIC:              makeIC(spec(eps(lbs(operatorv1.ExternalLoadBalancer, &managedDNS))), status(elbWithNullParameters())),
+			domainMatchesBaseDomain: true,
+		},
+		{
+			name:                    "loadbalancer type changed from Classic LB to unset, with NLB as default",
+			ic:                      makeIC(spec(eps(lbs(operatorv1.ExternalLoadBalancer, &managedDNS))), status(elb())),
+			ingressConfig:           ingressConfigWithDefaultNLB,
+			expectedResult:          true,
+			expectedIC:              makeIC(spec(eps(lbs(operatorv1.ExternalLoadBalancer, &managedDNS))), status(nlb())),
+			domainMatchesBaseDomain: true,
+		},
```

Comment:
Thinking about this further, what this test case shows is that if I have a cluster on AWS with an ingresscontroller that has a null `spec.endpointPublishingStrategy.loadBalancer.providerParameters` value and I do the following:
```shell
oc patch ingresses.config.openshift.io/cluster -p '{"spec":{"loadbalancer":{"platform":{"type":"AWS","aws":{"type":"NLB"}}}}}'
```
then the operator will change my ingresscontroller from a classic ELB to an NLB, right?

Reply:
yes

Reply:
Once we patch ingress config to NLB, the new ingress controller(not the existing running default IC or any other existing running IC) with nil ProviderParams will be created with NLB, however the existing running default IC and any other running exisiting IC until it is deleted will be Classic because the logic for recreating is in operatot.go